src/modules/fw_rate_control/FixedwingRateControl.cpp

```diff
@@ -404,6 +425,15 @@ void FixedwingRateControl::Run()
 		_actuator_controls.control[actuator_controls_s::INDEX_YAW] += _param_fw_rll_to_yaw_ff.get()
 				* constrain(_actuator_controls.control[actuator_controls_s::INDEX_ROLL], -1.0f, 1.0f);
 
+		// Tailsitter: rotate back to body frame from airspeed frame
+		if (_vehicle_status.is_vtol_tailsitter) {
+			const float helper = _actuator_controls.control[actuator_controls_s::INDEX_ROLL];
+			_actuator_controls.control[actuator_controls_s::INDEX_ROLL] =
+				_actuator_controls.control[actuator_controls_s::INDEX_YAW];
+
+			_actuator_controls.control[actuator_controls_s::INDEX_YAW] = -helper;
+		}
```

Comment:
The conversion seems to be repeatedly applied - Could we make the frame conversion a method, so that it can be consistently applied?

Reply:
Hmm I'm not that sure. We currently do the transformation both in the FW attitude and FW rate controller, which don't share a common, controller specific library. And duplicating the methods for both controllers is hardly an improvement to just do the conversion separately every time.